src/Rule/Integer.php

```diff
@@ -32,16 +32,36 @@ class Integer extends Rule
     ];
 
     /**
+     * A bool denoting whether or not strict checking should be done.
+     *
+     * @var bool
+     */
+    private $strict;
+
+    /**
+     * @param bool $strict
+     */
+    public function __construct($strict = false)
+    {
+        $this->strict = $strict;
+    }
+
+    /**
      * Validates if $value represents an integer.
      *
      * @param mixed $value
      * @return bool
      */
     public function validate($value)
     {
-        if (false !== filter_var($value, FILTER_VALIDATE_INT)) {
+        if ($this->strict) {
+            if (is_int($value)) {
+                return true;
+            }
+        } else if (false !== filter_var($value, FILTER_VALIDATE_INT)) {
```

Comment:
The nesting of ifs can be reduced here:

```php
        if ($this->strict && is_int($value)) {
            return true;
        }

        if (!$this->strict && filter_var($value, FILTER_VALIDATE_INT) !== false) {
            return true;
        }

        return $this->error(self::NOT_AN_INTEGER);
```

That's easier to read if you ask me, what do you think?